test/cppsuite/tests/session_microbenchmarks.cpp

```diff
@@ -0,0 +1,230 @@
+ */
+class session_microbenchmarks : public test {
+public:
+    session_microbenchmarks(const test_args &args) : test(args)
+    {
+        init_operation_tracker(
+          new operation_tracker_session_microbenchmarks(_config->get_subconfig(OPERATION_TRACKER),
+            _config->get_bool(COMPRESSION_ENABLED), *_timestamp_manager));
+    }
+
+    void
+    run() override final
+    {
+        /* You can remove the call to the base class to fully customize your test. */
+        test::run();
+    }
+
+    void
+    background_compact_operation(thread_worker *) override final
+    {
+        logger::log_msg(LOG_WARN, "background_compact_operation: nothing done");
+    }
+
+    void
+    checkpoint_operation(thread_worker *) override final
+    {
+        logger::log_msg(LOG_WARN, "checkpoint_operation: nothing done");
+    }
+
+    void
+    custom_operation(thread_worker *tc) override final
+    {
+        WT_SESSION wt_session = *(tc->session);
+        instruction_counter begin_transaction_counter(
+          "begin_transaction_instructions", test::_args.test_name);
+        instruction_counter commit_transaction_counter(
+          "commit_transaction_instructions", test::_args.test_name);
+        instruction_counter rollback_trancation_counter(
+          "rollback_trancation_instructions", test::_args.test_name);
+        instruction_counter prepare_transaction_counter(
+          "prepare_transaction_instructions", test::_args.test_name);
+
+        instruction_counter commit_after_prepare_transaction_counter(
+          "commit_after_prepare_instructions", test::_args.test_name);
+        instruction_counter open_cursor_cached_counter(
+          "open_cursor_cached_instructions", test::_args.test_name);
+        instruction_counter open_cursor_uncached_counter(
+          "open_cursor_uncached_instructions", test::_args.test_name);
+        instruction_counter timestamp_transaction_uint_counter(
+          "timestamp_transaction_uint_instructions", test::_args.test_name);
+        std::string cursor_uri = tc->db.get_collection(0).name;
+
+        /* Get the collection to work on. */
+        testutil_assert(tc->collection_count == 1);
+
+        scoped_session &session = tc->session;
+
+        begin_transaction_counter.track([&wt_session, &session]() -> int {
+            session->begin_transaction(session.get(), NULL);
+            return 0;
+        });
+
+        make_insert(tc, "1");
+
+        commit_transaction_counter.track([&wt_session, &session]() -> int {
+            session->commit_transaction(session.get(), NULL);
+            return 0;
+        });
+
+        session->begin_transaction(session.get(), NULL);
+
+        make_insert(tc, "2");
+
+        rollback_trancation_counter.track([&wt_session, &session]() -> int {
+            session->rollback_transaction(session.get(), NULL);
+            return 0;
+        });
+
+        std::string prepare_timestamp = tc->tsm->decimal_to_hex(tc->tsm->get_next_ts());
```

Comment:
We could use a single byte timestamp here but I don't know how much that would affect the number of instructions.

Reply:
When I tried that it said the timestamp wasn't big enough (that it needed to be larger than all the previous timestamps, I can track down the error if that helps).

Reply:
Oh of course because the populate stage is using MongoDB style timestamps.